### PR TITLE
Add support for configurable job uniqueness.

### PR DESCRIPTION
**Pull request by jpwhite4**
This is needed for proper job array support see also pull requests in the xdmod and xdmod-supremm and xdmod-xsede code base.

### CODE REVIEW

**Comment by connersaeli** [comment]
xdmod: https://github.com/ubccr/xdmod/pull/1723
xdmod-supremm: https://github.com/ubccr/xdmod-supremm/pull/327
xdmod-xsede: https://github.com/ubccr/xdmod-xsede/pull/400

**Review by connersaeli — CHANGES_REQUESTED** [review]

File: config/config.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
             "hostname_mode": "hostname",
             "pcp_log_dir": "/data/pcp-logs/my_cluster_name",
             "script_dir": "/data/jobscripts/my_cluster_name",
+            "job_uniq_mode": "local_job_id",
```
[the remark this answers is not in the document]
Comment (jpwhite4) [reply config/config.json]: Ok. Plan is to autodetect the job unique identifier - no need for a configuration setting.

File: src/supremm/xdmodaccount.py [hunk 2]
```diff
@@ -31,14 +24,14 @@ def __init__(self, resource_id, config):
...
-                self._query += "COALESCE(jf.`local_job_id_raw`, jf.`local_jobid`) as `job_uniq_id`, "
-            else:
-                self._query += "IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`)) as job_uniq_id, "
-
-            self._query += """
+                    CASE
+                      WHEN jf.`local_job_id_raw` IS NULL THEN IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                      WHEN sj.`source_format` = 'slurm' THEN jf.`local_job_id_raw`
+                      ELSE IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                    END AS job_uniq_id,
```
Comment (connersaeli) [inline src/supremm/xdmodaccount.py:31]: Does `job_uniq_id` need to be \``job_uniq_id`\` (with backticks)? Or is that a bit pedantic? Just thinking if it should match the syntax for the other aliases.

File: src/supremm/xdmodaccount.py [hunk 3]
```diff
@@ -84,14 +77,14 @@ def __init__(self, resource_id, config):
...
-                self._query += "COALESCE(jf.`local_job_id_raw`, jf.`local_jobid`) as `job_uniq_id`, "
-            else:
-                self._query += "IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`)) as job_uniq_id, "
-
-            self._query += """
+                    CASE
+                      WHEN jf.`local_job_id_raw` IS NULL THEN IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                      WHEN sj.`source_format` = 'slurm' THEN jf.`local_job_id_raw`
+                      ELSE IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                    END AS job_uniq_index,
```
Comment (connersaeli) [inline src/supremm/xdmodaccount.py:84]: Same comment here as line 31.

**Review by connersaeli — APPROVED** [review]

**Pull request merged**